File: webappdevelopment_ui/src/services/CustomerService.js
import http from "../http-common"; 

class CustomerService {
  getAllCustomers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/customer/customers`, searchDTO);
  }

  get(customerId) {
    return this.getRequest(`/customer/${customerId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/customer?field=${matchData}`, null);
  }

  addCustomer(data) {
    return http.post("/customer/addCustomer", data);
  }

  update(data) {
  	return http.post("/customer/updateCustomer", data);
  }
  
  uploadImage(data,customerId) {
  	return http.postForm("/customer/uploadImage/"+customerId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CustomerService();

File: src/main/java/com/webappdevelopment/service/CustomerService.java
package com.webappdevelopment.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.webappdevelopment.domain.Customer;
import com.webappdevelopment.dto.CustomerDTO;
import com.webappdevelopment.dto.CustomerSearchDTO;
import com.webappdevelopment.dto.CustomerPageDTO;
import com.webappdevelopment.dto.CustomerConvertCriteriaDTO;
import com.webappdevelopment.service.GenericService;
import com.webappdevelopment.dto.common.RequestDTO;
import com.webappdevelopment.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CustomerService extends GenericService<Customer, Integer> {

	List<Customer> findAll();

	ResultDTO addCustomer(CustomerDTO customerDTO, RequestDTO requestDTO);

	ResultDTO updateCustomer(CustomerDTO customerDTO, RequestDTO requestDTO);

    Page<Customer> getAllCustomers(Pageable pageable);

    Page<Customer> getAllCustomers(Specification<Customer> spec, Pageable pageable);

	ResponseEntity<CustomerPageDTO> getCustomers(CustomerSearchDTO customerSearchDTO);
	
	List<CustomerDTO> convertCustomersToCustomerDTOs(List<Customer> customers, CustomerConvertCriteriaDTO convertCriteria);

	CustomerDTO getCustomerDTOById(Integer customerId);







}





